hyperspy/io_plugins/emd.py

```diff
@@ -828,13 +828,18 @@ def _get_metadata_dict(self, om):
         meta_gen = {}
         meta_gen['original_filename'] = os.path.split(self.filename)[1]
         meta_gen['title'] = self.detector_name
-        # We have only one entry in the original_metadata, so we can't use the
-        # mapping of the original_metadata to set the date and time in the
-        # metadata: need to set manually here
-        unix_time = om['Acquisition']['AcquisitionStartDatetime']['DateTime']
-        date, time = self._convert_datetime(unix_time).split('T')
-        meta_gen['date'] = date
-        meta_gen['time'] = time
+        try:
+            # We have only one entry in the original_metadata, so we can't use the
+            # mapping of the original_metadata to set the date and time in the
+            # metadata: need to set manually here
+            unix_time = om['Acquisition']['AcquisitionStartDatetime']['DateTime']
+            date, time = self._convert_datetime(unix_time).split('T')
+            meta_gen['date'] = date
+            meta_gen['time'] = time
+        except KeyError:
```

Comment:
@robindegen, if the `AcquisitionStartDatetime` key is not present in the metadata of TEM images, do you know how we can parse the date and time?

Reply:
@ericpre As far as I can tell, AcquisitionStartDatetime should have been filled in for TEM images. It might however be dependent on the TEM server version that Velox is running on. I'd have to discuss internally why these values could be missing. Do you have any EMD files that you could share that have this issue?

What you can do however is  use ```CustomProperties``` which has a bunch of key-value pairs. One of them is ```Detectors[BM-Ceta].TimeStamp``` which should contain the timestamp for that particular frame. The only downside is is that this is not the time the acquisition was started, but rather then that particular frame was stored.

Reply:
Thanks for the quick reply. I have shared two files on [dropbox](https://www.dropbox.com/sh/50mopwcbtk1ea0k/AADZsKCD3zaQVhBB-zTUHgqUa), corresponding to different versions of velox and TEM server. None of them has got the `AcquisitionStartDatetime` metadata but the most recent one has `Detectors[BM-Ceta].TimeStamp`. Sorry, I don't know the TEM server version on top of my head ;), but I can check later.